js/app.js

```diff
@@ -126,14 +126,13 @@ function randomizeClearings(gameMap) {
   return availableClearings
     .map(clearing => ({ sortIndex: Math.random(), value: clearing }))
     .sort((a, b) => a.sortIndex - b.sortIndex)
-    .map(sortableClearing => sortableClearing.value)
-    .splice(0, gameMap.numClearings);
+    .map(sortableClearing => sortableClearing.value);
 }
 
 function randomizeMap() {
   const map = DATA.MAP_LIST[Math.floor(Math.random() * DATA.MAP_LIST.length)];
   return {
-    name: map.name,
+    ...map,
```

Comment:
I hesitate somewhat to leak out the map from data as the actual logical map that the "view" portion of the code uses to generate HTML. With the current fields it doesn't really matter, but, it's interesting to think about separation of concerns and where to impose strict contracts vs passing through data. Passing through the data couples HTML generation directly with the "database", for better or worse. I'm curious to hear what you think.

Reply:
Yea I agree overall. This is where typing information would have been really nice as I added the fields in `data.js`, but they did not show up in the generateMapHtml() function (a compiler would have immediately thrown rather than let me get `undefined`). 

It seems from this comment and the other one that we both want a better mechanism to isolate these concepts. I think this will be refactored as part of that rework and am ok with keeping it as-is in this PR. Unless you have an immediate way to improve it here?

Reply:
I would just not dump all data fields into the return object here. In other words, don't use the "..." operator and instead explicitly expose the fields you need. Typing would definitely make these interfaces more clear.

Reply:
Done. A bit weird since we need every field lol 😆